### DISCUSSION TITLE
Always show root dir

### DISCUSSION THREAD

**Discussion by jrock2004** [nvim-neo-tree/neo-tree.nvim#963] [Q&A]
So if I open a file in a sub folder and open Neo-tree, the only folder it showing is the current one I am in. How can I change this? Here is my config?

```
return {
  "nvim-neo-tree/neo-tree.nvim",
  opts = {
    close_if_last_window = true,
    enable_git_status = true,
    enable_diagnostics = true,
    filesystem = {
      filtered_items = {
        hide_dotfiles = false,
      },
      follow_current_file = false,
    },
    window = {
      mappings = {
        ["<C-x>"] = "open_split",
        ["<C-v>"] = "open_vsplit",
      },
    },
  },
}
```

**Comment by cseickel** [comment]
It will use your current working directory of the tab by default if you open in a sidebar, or the cwd of the window if you open netrw style. Do you have another plugin that is automatically setting your CWD to that of the file?

**Comment by dtasada** [comment]
Did you ever find the fix? I have this issue. I always shows a few folders above, but constantly changes the root directory when i switch files.

**Reply by cseickel** [reply]
You must have `autochdir` turned on or a plugin like rooter that automatically changes your cwd.
https://neovim.io/doc/user/options.html#'autochdir'

Also see:
```
:h neo-tree-cwd 
```

**Comment by jrock2004** [comment]
For anyone who needs a solution for this, I finally took the time to fix this. So when I toggle neotree this is the code I execute

```lua
  local wk = require 'which-key'

  wk.register {
    ['<leader>e'] = {
      function()
        local lspconfig = require 'lspconfig'
        local root_patterns = { '.git' }
        local root_dir = lspconfig.util.root_pattern(unpack(root_patterns))(vim.fn.expand '%:p')

        require('neo-tree.command').execute { toggle = true, dir = root_dir }
      end,
      'Explorer',
    },
  }
```
